pkg/dcmtime/example_test.go

```diff
@@ -0,0 +1,465 @@
+	fmt.Println("STRING:", tm.String())
+
+	// Output:
+	// DCM   : 150405.431
+	// STRING: 15:04:05.431
+}
+
+func ExampleTime_precisionMinutes() {
+	// We'll use the reference date as our date
+	timeVal, err := time.Parse(
+		"15:04",
+		"15:04",
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create a nw TM object like so:
+	tm := Time{
+		Time:      timeVal,
+		Precision: PrecisionMinutes,
+	}
+
+	// Get the DICOM string value
+	fmt.Println("DCM   :", tm.DCM())
+
+	// Our String() method will yield a more readable non-DICOM-compliant value.
+	fmt.Println("STRING:", tm.String())
+
+	// Output:
+	// DCM   : 1504
+	// STRING: 15:04
+}
+
+// Parse a datetime string.
+func ExampleParseDatetime() {
+	// This is a DT value like we would expect
+	daString := "20201210123001.000431+0100"
+
+	// We are parsing the date string without allowing nema
+	dt, err := ParseDatetime(daString)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("TIME VALUE:", dt.Time)
+	fmt.Println("PRECISION :", dt.Precision)
+	fmt.Println("HAS OFFSET:", dt.NoOffset)
+
+	// Output:
+	// TIME VALUE: 2020-12-10 12:30:01.000431 +0100 +0100
+	// PRECISION : FULL
+	// HAS OFFSET: true
+}
+
+// Parse a datetime string with no timezone.
+func ExampleParseDatetime_noTimezone() {
+	// This is a DT value like we would expect
+	daString := "20201210123001.000431"
+
+	// We are parsing the date string without allowing nema
+	dt, err := ParseDatetime(daString)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("TIME VALUE:", dt.Time)
+	fmt.Println("PRECISION :", dt.Precision)
+	fmt.Println("HAS OFFSET:", dt.NoOffset)
+
+	// Output:
+	// TIME VALUE: 2020-12-10 12:30:01.000431 +0000 +0000
+	// PRECISION : FULL
+	// HAS OFFSET: false
+}
+
+// Parse a datetime string with no timezone.
+func ExampleParseDatetime_precisionHour() {
+	// This is a DT value like we would expect
+	daString := "2020121012"
+
+	// We are parsing the date string without allowing nema
+	dt, err := ParseDatetime(daString)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("TIME VALUE:", dt.Time)
+	fmt.Println("PRECISION :", dt.Precision)
+	fmt.Println("HAS OFFSET:", dt.NoOffset)
+
+	// Output:
+	// TIME VALUE: 2020-12-10 12:00:00 +0000 +0000
+	// PRECISION : HOURS
+	// HAS OFFSET: false
+}
+
+func ExampleDatetime() {
+	// We'll use the reference date as our date
+	timeVal, err := time.Parse(
+		"2006-01-02T15:04:05.000000-07:00",
+		"2006-01-02T15:04:05.123456+01:00",
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create a nw TM object like so:
+	dt := Datetime{
+		Time:      timeVal,
+		Precision: PrecisionFull,
+		NoOffset:  false,
+	}
+
+	// Get the DICOM string value
+	fmt.Println("DCM   :", dt.DCM())
+
+	// Our String() method will yield a more readable non-DICOM-compliant value.
+	fmt.Println("STRING:", dt.String())
+
+	// Output:
+	// DCM   : 20060102150405.123456+0100
+	// STRING: 2006-01-02 15:04:05.123456 +01:00
+}
+
+func ExampleDatetime_noOffset() {
```

Comment:
I'm actually of the opposite opinion -- I think more code examples are almost always better. When I'm consuming a library, I almost always tend to flip through the examples first and only dig into the API docs if I have to.

I tend to think of FastAPI's documentation as kind of the gold standard: https://fastapi.tiangolo.com/tutorial/, where you don't have to just sift through API descriptions to figure our the lib -- It's laid out nicely for you. there are clear (and multiple) examples for almost every part of the library.

Granted, FastAPI is an enormous library that really does need this sort of manual book approach, but I think that DICOM dates have enough little quirks that having.

Since examples are collapsed by default, I don't think having a few examples per data type is going to clutter the docs too badly.

That being said, I would like to defer to your preference, so if you want to select some examples to remove I would be happy to remove them!